template/node/index.js

```diff
@@ -8,24 +8,31 @@ const getStdin = require('get-stdin');
 const handler = require('./function/handler');
 
 getStdin().then(val => {
-    handler(val, (err, res) => {
+    const cb = (err, res) => {
         if (err) {
             return console.error(err);
         }
-        if(isArray(res) || isObject(res)) {
+        if (!res) {
+            return;
+        }
+        if(Array.isArray(res) || isObject(res)) {
             console.log(JSON.stringify(res));
         } else {
             process.stdout.write(res);
         }
-    });
+    } // cb ...
+
+    const result = handler(val, cb);
+    if (result instanceof Promise) {
```

Comment:
You're missing a branch where the result is _not_ a `Promise`. I think that will break backwards compatibility where someone updates their existing function to the new template or they write a new function the old way expecting it to still work.

Reply:
You are right. I missed that part. I'll update code.

Reply:
I don't think so. The portion of code responsible of returning the result of the function is now packet in `cb`, which is passed to `handler`. If you are running a "legacy" function `cb` will be called, the result sent to the GW and all ends there.
But if in turn you are running a function that returns a Promise, it will be "unresolved" in `index.js` and that's why we check if it's a Promise, and resolve it to return a result.